src/host/getset.cpp

```diff
@@ -589,7 +589,15 @@ void ApiRoutines::GetLargestConsoleWindowSizeImpl(const SCREEN_INFORMATION& cont
         if (NewSize.X != context.GetViewport().Width() ||
             NewSize.Y != context.GetViewport().Height())
         {
+            // GH#1856 - make sure to hide the commandline _before_ we execute
+            // the resize, and the re-display it after the resize. If we leave
+            // it displayed, we'll crash during the resize when we try to figure
+            // out if the bounds of the old commandline fit within the new
+            // window (it might not).
+            CommandLine& commandLine = CommandLine::Instance();
+            commandLine.Hide(FALSE);
             context.SetViewportSize(&NewSize);
+            commandLine.Show();

```

Comment:
does this have an impact on any traditional console interactive scenarios? I'm a bit worried about that.

Reply:
Oh -- resizing goes through here before because it's coming off the pty's resize handle? Does a user-initiated window resize on conhost run through ApiRoutines too?

Reply:
I'm pretty OK with this because the console host used to do this a ton in the past. It always hid the command line display before manipulating anything and then restored it when it was done. It's kinda sad that we have to do it again, but if it prevents a crash, I'm not that bothered by it.